controllers/reconcilers/serviceaccount_reconciler.go

```diff
@@ -72,26 +71,26 @@ func (r *ModelMeshServiceAccountReconciler) Reconcile(ctx context.Context, log l
 	return nil
 }
 
-func (r *ModelMeshServiceAccountReconciler) Cleanup(ctx context.Context, log logr.Logger, isvcNs string) error {
+func (r *ServiceAccountReconciler) Cleanup(ctx context.Context, log logr.Logger, isvcNs string) error {
 	log.V(1).Info("Deleting ServiceAccount object for target namespace")
-	return r.serviceAccountHandler.DeleteServiceAccount(ctx, types.NamespacedName{Name: modelMeshServiceAccountName, Namespace: isvcNs})
+	return r.serviceAccountHandler.DeleteServiceAccount(ctx, types.NamespacedName{Name: r.serviceAccountName, Namespace: isvcNs})
```

Comment:
Looks like you've changed the KServe service account to  `default`... I don't think we should be deleting a ServiceAccount that we didn't create. Deleting the `default` SA can have an impact to the user.

Reply:
I'm not calling the serviceaccount reconciler from the raw inferenceservice reconciler so the `default` SA is not deleted